pinot-core/src/main/java/org/apache/pinot/core/periodictask/BasePeriodicTask.java

```diff
@@ -111,22 +126,47 @@ protected void setUpTask() {
    */
   @Override
   public final void run() {
-    _running = true;
-
-    if (_started) {
-      long startTime = System.currentTimeMillis();
-      LOGGER.info("Start running task: {}", _taskName);
-      try {
-        runTask();
-      } catch (Exception e) {
-        LOGGER.error("Caught exception while running task: {}", _taskName, e);
+    try {
+      // Don't allow a task to run more than once at a time.
+      _runLock.lock();
+      _running = true;
+
+      String periodicTaskRequestId =
+          _activePeriodicTaskProperties != null ? _activePeriodicTaskProperties.getProperty(PeriodicTask.PROPERTY_KEY_REQUEST_ID) : null;
+      if (_started) {
+        long startTime = System.currentTimeMillis();
+        LOGGER.info("[TaskRequestId: {}] Start running task: {}", periodicTaskRequestId, _taskName);
```

Comment:
If periodicTaskId is null, we will see the log as `[TaskRequestId: null]`. Not the best experience. Maybe set the requireId to be some specific value ("automatic" or "periodic execution" or some such, maybe?)

Reply:
Fixed.